README.md

```diff
@@ -2,21 +2,123 @@
 
 Aim: Implement algorithms to find the global minima of the Eggholder function and Shekel function exploiting parallel architectures in an orderly C++ way.
 
-## C++ Software Structure 
-- [ ]  Use of Eigen or Armadillo?
-- [ ] Intel MKL for distributed linear algebra  
+> Repository for Maxwell Institute Advanced Course: High Performance Computing for Mathematicians
+
+## Getting Started
+
+These implementations use [Eigen 3.3.9](https://eigen.tuxfamily.org/index.php?title=Main_Page) for its linear algebra. I place Eigen in the root directory – it is ignored on pushing (See `.gitignore`). Then to compile the file `gd.cpp` with Eigen using g++,
+
+```
+g++ -I .\eigen-3.3.9 GradientDescent\gd.cpp -o gd.exe
+```
```

Comment:
This needs to be g++ -I ./eigen-3.3.9 GradientDescent/gd.cpp -o gd.exe for me (with "/" instead of "\") but maybe this is a windows/linux issue